src/main/java/io/pravega/connectors/flink/ReaderCheckpointHook.java

```diff
@@ -97,7 +95,17 @@ public String getIdentifier() {
         // we make sure the executor is shut down after the future completes
         checkpointResult.handle((success, failure) -> scheduledExecutorService.shutdownNow());
 
-        return flinkFutureFromJava8Future(checkpointResult, executor);
+        // handle checkpoint completion
```

Comment:
The `handleAsync` code here does nothing. In English it says "when the checkpointResult completes, complete the checkpointResult".